tests/integration/test_charm.py

```diff
@@ -362,3 +410,14 @@ async def test_can_connect_exporter_and_get_metrics(self, ops_test: OpsTest):
         assert 'mlflow_metric{metric_name="num_experiments"} 2.0' in metrics_text
         assert 'mlflow_metric{metric_name="num_registered_models"} 0.0' in metrics_text
         assert 'mlflow_metric{metric_name="num_runs"} 0' in metrics_text
+
+    @pytest.mark.abort_on_fail
+    async def test_ingress_relation(self, ops_test: OpsTest):
+        """Setup Istio and relate it to the Tensoboards Web App(TWA)."""
+        await setup_istio(ops_test, ISTIO_GATEWAY_CHARM_NAME, ISTIO_PILOT_CHARM_NAME)
+
+        await ops_test.model.add_relation(
+            f"{ISTIO_PILOT_CHARM_NAME}:ingress", f"{CHARM_NAME}:ingress"
+        )
+
+        await ops_test.model.wait_for_idle(apps=[CHARM_NAME], status="active", timeout=60 * 5)
```

Comment:
can we add a check to see if we can actually get to the mlflow page through the ingress?

Reply:
I think you meant to include this check but it got missed?  I see fetch_response and url_with_ingress are unused